test/goldens/end2end/_dummy_pkg-1.0.0-null-safety.1.json_report.md

```diff
@@ -2,6 +2,15 @@
 
 ### [x] 0/10 points: Provide a valid `pubspec.yaml`
 
+<details>
+<summary>
+Failed to verify repository URL.
+</summary>
+
+Please provide a valid repository URL.
+
+`pkg/pub_integration/test_data/_dummy_pkg/pubspec.yaml` from the repository has no `version`.
+</details>
```

Comment:
Look at this, I think we should reorder it, so things like bad SDK constraints comes before repository verification.

I also think we need to improve the message a bit.

We could say things like:
 * Found repository URL `https://github.com/dart-lang/pub-dev` in `homepage` property of `pubspec.yaml`.
 * Cloned repository and found `pkg/pub_integration/test_data/_dummy_pkg/pubspec.yaml` with the correct package name, but no `version` property, this it is unlikely this package was published from this repository.